File: 2737-row-with-maximum-ones/row-with-maximum-ones.scala
object Solution {
    def rowAndMaximumOnes(mat: Array[Array[Int]]): Array[Int] = {
        def iter(matrix: Array[Array[Int]], index: Int, result: Array[Int]): Array[Int] =
            matrix.headOption match
                case Some(column) =>
                    val rowSum = column.sum
                    val updatedResult = if rowSum > result(1) then Array(index, rowSum) else result
                    iter(matrix.tail, index + 1, updatedResult)
                case None         => result

        iter(mat, 0, Array(0,0))
    }
}

File: 2737-row-with-maximum-ones/row-with-maximum-ones.java
class Solution {
    public int[] rowAndMaximumOnes(int[][] mat) {
        int maxOnesIdx = 0;
        int maxOnesCount = 0;
        for(int i = 0; i < mat.length; i++){
            int onesCount = 0;

            for(int j = 0; j < mat[i].length; j++){
                onesCount += mat[i][j];
            }

            if(onesCount > maxOnesCount){
                maxOnesIdx = i;
                maxOnesCount = onesCount;
            }
        }
        return new int[]{maxOnesIdx, maxOnesCount};
    }
}
